### PR TITLE
CVE-2012-4387

### PR DESCRIPTION
**Pull request by based-avocado**
Provide additional information on CVE-2012-4387 via its corresponding `.yml` file.

### CODE REVIEW

File: cves/CVE-2012-4387.yml [hunk 1]
```diff
@@ -6,33 +6,33 @@ nickname_instructions: |
...
   alongside its CVE to make it more easily distinguished from the rest.
-nickname:
+nickname: Long Parameter Names DoS Opportunity
 
 CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE:
+CWE: (CWE-641) Improper Restriction of Names for Files and Other Resources
```
Comment (renchauret) [inline cves/CVE-2012-4387.yml]: Should just be the integer "641"

File: cves/CVE-2012-4387.yml [hunk 2]
```diff
@@ -60,7 +65,7 @@ fixes_vcc_instructions: |
   Please put the SVN commit number in "commit" below, and any notes about how this
   was discovered in the "note" field.
 fixes:
-  - commit: 87935af56a27235e9399308ee1fcfb74f8edcefa
+  - commit: 80e03182d66d9e6ab18f9a9a9b3c42725a1c89e9
     note: Restricted parameter length
```
Comment (renchauret) [inline cves/CVE-2012-4387.yml:69]: Does this note still apply after changing the fix commit?

File: cves/CVE-2012-4387.yml [hunk 3]
```diff
@@ -119,20 +128,25 @@ discovered:
     The "apache" flag can be true, false, or nil.
     If there is no evidence as to how this vulnerability was found, then you may
     leave the entries blank except for "answer". Write down where you looked in "answer".
-  answer:
+  answer: |
+    From what I can find on the issue itself, this bug was found (and patched) by Johno
+    Crawford, who's some guy who works for an Italian software company, Sulake. This is
```
Comment (renchauret) [inline cves/CVE-2012-4387.yml]: Perhaps rephrase "some guy"

File: cves/CVE-2012-4387.yml [hunk 4]
```diff
@@ -205,4 +231,10 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    This was caused by having user inputs be translated to complex expressions with
```
Comment (renchauret) [inline cves/CVE-2012-4387.yml]: Does this mean this was a flaw in the design?

**Comment by renchauret** [comment]
2 upvotes

File: cves/CVE-2012-4387.yml (hunk 1 above)
Comment (randallkentbrown) [reply cves/CVE-2012-4387.yml]: Agree with Ren.

File: cves/CVE-2012-4387.yml (hunk 3 above)
Comment (randallkentbrown) [reply cves/CVE-2012-4387.yml]: Agree with Ren.

File: cves/CVE-2012-4387.yml [hunk 5]
```diff
@@ -192,8 +214,12 @@ lessons:
     applies:
     note:
   complex_inputs:
-    applies:
-    note:
+    applies: true
+    note: |
+      Similar to the above point on not trusting inputs, perhaps it would've been
+      wiser from an architectural perspective to make requests not directly translate
```
Comment (randallkentbrown) [inline cves/CVE-2012-4387.yml:221]: Does this mean that another approach would have been better, or does the team's OGNL code just more thorough input validation?

File: cves/CVE-2012-4387.yml (hunk 2 above)
Comment (based-avocado) [reply cves/CVE-2012-4387.yml:69]: It does indeed! The fix itself was restricting the parameter length, but the commit hash for this fix was not originally pointing to the correct commit.

File: cves/CVE-2012-4387.yml (hunk 1 above)
Comment (based-avocado) [reply cves/CVE-2012-4387.yml]: Got it, will change in a bit.

File: cves/CVE-2012-4387.yml (hunk 5 above)
Comment (based-avocado) [reply cves/CVE-2012-4387.yml:221]: I'm don't think that the choice to use OGNL was necessarily bad, but it's important to understand the inherent limitations and vulnerabilities present in any third-party technology you opt to use in your project. In this case, that means more thorough input validation.

File: cves/CVE-2012-4387.yml (hunk 4 above)
Comment (based-avocado) [reply cves/CVE-2012-4387.yml]: I would classify this more as a coding mistake than a design flaw. The approach they opted to take was fine, but they should have been more careful with validation the input they were receiving. Updating my answer to better convey this information.

**Comment by randallkentbrown** [comment]
2 upvotes

**Pull request merged**